utopia-remix/app/routes/v1.project.$id.tsx

```diff
@@ -20,8 +23,47 @@ export async function loader(args: LoaderFunctionArgs) {
 export async function action(args: ActionFunctionArgs) {
   return handle(args, {
     PUT: {
+      handler: createProject,
+      validator: ALLOW,
+    },
+    POST: {
       handler: proxy,
       validator: ALLOW,
     },
   })
 }
+
+async function createProject(req: Request, params: Params<string>) {
+  const { id } = params
+  ensure(id != null, 'project id is null', Status.BAD_REQUEST)
+
+  // create the project
+  const project = await proxy(req)
+
+  // handle access level setting
+  const url = new URL(req.url)
+  const accessLevelParam = url.searchParams.get('accessLevel')
+  const accessLevel: AccessLevel | null = accessLevelParamToAccessLevel(accessLevelParam)
+  if (accessLevel != null) {
+    await setProjectAccess({ projectId: id, accessLevel: accessLevel })
+  }
+
+  return project
+}
```

Comment:
I think this should _always_ create the project access, with a default to `PRIVATE`. A project should _always_ have an access (and we should address the nullability separately, but the concept stands).
Of course if the `accessLevel` param is passed, it overrides the default (if it has a valid value).

Reply:
ok, will do. It means though that if someone is calling PUT on a public project by accident (without setting access level) it will be reverted back to PRIVATE

Reply:
I don't think we should – the call should be an upsert with a `DO NOTHING` if the access level already exists, because we have separate endpoints for changing the access level (this is just metadata bootstrapping)

Reply:
changed it to default to PRIVATE and on conflict do nothing